lib/hive/controller/android.rb

```diff
@@ -23,7 +26,7 @@ def get_connected_devices
         unless Hive.hive_mind.device_details.has_key?(:error)
           # Selecting only android mobiles
           begin
-            connected_devices = Hive.hive_mind.device_details['connected_devices'].select{ |d| d['device_type'] == 'Mobile' && d['operating_system_name'] == 'android' }
+            connected_devices = Hive.hive_mind.device_details['connected_devices'].select{ |d| d['device_type'] == @device_type && d['operating_system_name'] == 'android' }
```

Comment:
I haven't changed this part of the code from the previous version.. I don't think we have any problems with devices being updates, but I think possibly this could cause a problem if we use the OS version attribute in the device object for anything elsewhere in hive runner (I don't know if we do), as it seems to get the OS version attribute from Hivemind rather than an internal version.